users/tests.py

```diff
@@ -1,3 +1,64 @@
-from django.test import TestCase
+import jwt
 
-# Create your tests here.
+from unittest.mock import patch, MagicMock
+
+from django.test import TestCase, Client
+from django.conf import settings
+
+from users.views import KakaoClient
+
+client = Client()
+class KakaoLoginTest(TestCase):
+    @patch.object(KakaoClient, "request_access_token")
+    def test_kakao_auth_code_invalid(self, mocked_token):
+        class MockedToken:
+            def json(self):
+                return {"no_access_token" : "no_access_token"}
+        
+        mocked_token.return_value = MockedToken().json()
```

Comment:
상안님 메소드를 통째로 모킹하는 경우엔 굳이 Mocking 클래스를 만드실 필요가 없습니다. 어차피 반환되는 값만 정해주면 되기 때문이죠. 저희가 Mocking 클래스를 만들었던 이유는 해당 메소드 전부가 아닌 하나의 request 요청의 값만 모킹하기 위함입니다. 그래서 내부에 `json()` 메소드를 만들어 주어서 그 값이 일치하게 만들어준것이구요
```suggestion
        mocked_token.return_value = {"message" : "no_access_token"}
```